Documents/Pseudocode/Pseudocode_Final

```diff
@@ -101,8 +102,9 @@ Class Network
 									//just make another data member for validation data
 		//open previous_network_filename
 		//open validation_data_filename
-		//readInit(previous_network_filename)
-		//classify(validation_data_filename)
+		//readInit(previous_network_filename)	//This method creates the vector of matrices for sum_nabla_b and other matrics that are not needed during classification, consider creating the vector matrices needed inside of this function [-Ami]
+							// parameter not needed for classify () as the training_data_filename stores the name of the validation file [-Ami]
+							//classify is already being called by main() [-Ami]
```

Comment:
to clarify: no need for validation data file because I considered the training file name to be written into the previous network file, viz. readInit(file)

Reply:
I was thinking about the fact that classification only requires the forwardPropagation(). So, we won't be needing the sum_nabla_b and sum_nabla_w matrices for classification. So, let's assign them them with a 0 (though, I am still doubtful about this!)

Reply:
Did you look at the previous one, where I made a constructor specifically for the classification? It wasn't calling readInit() for classification. readInit(string &) gets called only when we want to train an existing network.

Reply:
That's true, but we had a function that would do everything for us, so why not use it? Setup happens once, so it wouldn't result in any major slowdown.

Initializing the nabla vectors with zeros only happens at setup. The first time the user runs backPropagation() it will total the results with any values already present. The nabla vectors are initialized to 0 for the same reason they are zeroed out after every mini-batch, during SGD.

I had taken your meaning to override the constructor for classification to be: let the user give us a validation file and a previous (trained) network, then the network will produce output -- with only one line of code. And, at the time, I thought we wanted a constructor that took a filename to set up an existing network, and the classification override in the previous version of pseudocode would have made that impossible, since the argument lists would have been the same (could be mistaken there). Additionally, having to run another constructor and train a whole network before using the classify constructor, and *then* run classify() just didn't seem to fit the description.

To my understanding, readinit(filename) gets called when you want to load/initialize a previous network from a file, regardless of whether you intend to train or classify.

Reply:
Forgot to mention: as a by-product of train(), a network's nabla vectors should only be non-zero during training. The upshot of this is that when we load a trained network, it should already have zeros in its nabla vectors. 

We could probably figure out ways to only load what we need, given the circumstances (i.e. which constructor we call), and you've pretty much got that covered. I would just make a minor request that for modularity's sake we defer the responsibility of initializing networks from files/cin to readInit() functions.

Reply:
I think we could do that, initializing the network matrices in readInit()! But, creating the matrices or initializing the parameters of the class usually happens in the constructor. readInit(), as the name suggests, is just responsible for reading in the values from the user.

Reply:
Actually, it doesn't matter where we create the matrices as such. So, I am okay with creating the required matrices in the readInit() methods.

Reply:
Sounds like a solid jurisdiction for readInit(): only to read contents of a file or cin. In practice, we might be able to pass a flag to signal to readInit() how much to take from the previous network file, or something. Otherwise, I think we can merge this branch after the edits we discussed are implemented.